source/family/ambiq/ama3b1kk/target_reset.c

```diff
@@ -0,0 +1,242 @@
+            }
+            if (!swd_write_word(MCUCTRL_SCRATCH0, (scratch0 | 0x1)))
+            {
+                debug_msg("if (!swd_write_word(MCUCTRL_SCRATCH0, (scratch0 | 0x1))) \r\n");
+                return 0;
+            }
+        }
+        else
+        {
+            if (!swd_write_word(DBG_EMCR, VC_CORERESET))
+            {
+                debug_msg("if (!swd_write_word(DBG_EMCR, VC_CORERESET)) \r\n");
+                return 0;
+            }
+        }
+
+        // Perform a soft reset
+        if (!swd_read_word(NVIC_AIRCR, &val))
+        {
+            debug_msg("if (!swd_read_word(NVIC_AIRCR, &val)) \r\n");
+            return 0;
+        }
+
+        if (!swd_write_word(NVIC_AIRCR, VECTKEY | (val & SCB_AIRCR_PRIGROUP_Msk) | SYSRESETREQ))
+        {
+            debug_msg("if (!swd_write_word(NVIC_AIRCR, VECTKEY | (val & SCB_AIRCR_PRIGROUP_Msk) | SYSRESETREQ)) \r\n");
+            return 0;
+        }
+
+        osDelay(10);
+
+        do
+        {
+            if (!swd_read_word(DBG_HCSR, &val))
+            {
+                debug_msg("if (!swd_read_word(DBG_HCSR, &val))  221\r\n");
+                return 0;
+            }
+        } while ((val & S_HALT) == 0);
+
+        // Disable halt on reset
+        if (secure)
+        {
+            if (!swd_write_word(MCUCTRL_SCRATCH0, scratch0))
+            {
+                debug_msg("if (!swd_write_word(MCUCTRL_SCRATCH0, scratch0)) \r\n");
+                return 0;
+            }
+        }
+        else
+        {
+            if (!swd_write_word(DBG_EMCR, 0))
+            {
+                debug_msg("if (!swd_write_word(DBG_EMCR, 0)) \r\n");
+                return 0;
+            }
+        }
+
+        break;
+
+    case NO_DEBUG:
+        debug_msg("swd_set_state_ama3b1kk NO_DEBUG \r\n");
+        if (!swd_write_word(DBG_HCSR, DBGKEY))
+        {
+            return 0;
+        }
+
+        break;
+
+    case DEBUG:
+        debug_msg("swd_set_state_ama3b1kk DEBUG \r\n");
+        if (!JTAG2SWD())
+        {
+            return 0;
+        }
+
+        if (!swd_write_dp(DP_ABORT, STKCMPCLR | STKERRCLR | WDERRCLR | ORUNERRCLR))
+        {
+            return 0;
+        }
+
+        // Ensure CTRL/STAT register selected in DPBANKSEL
+        if (!swd_write_dp(DP_SELECT, 0))
+        {
+            return 0;
+        }
+
+        // Power up
+        if (!swd_write_dp(DP_CTRL_STAT, CSYSPWRUPREQ | CDBGPWRUPREQ))
+        {
+            return 0;
+        }
+
+        // Enable debug
+        if (!swd_write_word(DBG_HCSR, DBGKEY | C_DEBUGEN))
+        {
+            return 0;
+        }
+
+        break;
+
+    default:
+        return 0;
+    }
+
+    return 1;
+}
+
+const target_family_descriptor_t g_ambiq_ama3b1kk = {
+    .family_id = kAmbiq_ama3b1kk_FamilyID,
+    //.default_reset_type = kHardwareReset,
+    .default_reset_type = kSoftwareReset,
+    .soft_reset_type = SYSRESETREQ,
+    //.target_before_init_debug = target_before_init_debug,
```

Comment:
why are all these commented out?

Reply:
These we reminders of things that were available. I removed them now.